### PR TITLE
[bitnami/minio] Introduced probes in gateway deployment

### PR DESCRIPTION
**Pull request by blafry**
Signed-off-by: Błażej Frydlewicz <b.frydlewicz@piwik.pro>

**Description of the change**

Ported existing solution of `livenessProbe` `readinessProbe` and `startupProbe` into gateway deployment.

**Benefits**

Probes are configured.

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/minio/templates/gateway/deployment.yaml [hunk 1]
```diff
@@ -167,6 +167,44 @@ spec:
             - name: minio-console
               containerPort: {{ .Values.containerPorts.console }}
               protocol: TCP
+          {{- if .Values.livenessProbe.enabled }}
+          livenessProbe:
```
Comment (fmulero) [inline bitnami/minio/templates/gateway/deployment.yaml:171]: If we apply this suggestion we could simplify the code
```suggestion
          livenessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.livenessProbe "enabled") "context" $) | nindent 12 }}
```

File: bitnami/minio/templates/gateway/deployment.yaml [hunk 2]
```diff
@@ -167,6 +167,44 @@ spec:
...
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+          {{- else if .Values.customLivenessProbe }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.customLivenessProbe "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.readinessProbe.enabled }}
+          readinessProbe:
```
Comment (fmulero) [inline bitnami/minio/templates/gateway/deployment.yaml:185]: Same as in `livenessProbe`

```suggestion
          readinessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.readinessProbe "enabled") "context" $) | nindent 12 }}
```

File: bitnami/minio/templates/gateway/deployment.yaml [hunk 3]
```diff
@@ -167,6 +167,44 @@ spec:
...
+            tcpSocket:
+              port: minio-api
+            initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.readinessProbe.successThreshold }}
+            failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
```
Comment (fmulero) [inline bitnami/minio/templates/gateway/deployment.yaml:192]: These lines can be removed

```suggestion
```

File: bitnami/minio/templates/gateway/deployment.yaml [hunk 4]
```diff
@@ -167,6 +167,44 @@ spec:
...
+          livenessProbe:
+            httpGet:
+              path: /minio/health/live
+              port: minio-api
+              scheme: {{ ternary "HTTPS" "HTTP" .Values.tls.enabled | quote }}
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
```
Comment (fmulero) [inline bitnami/minio/templates/gateway/deployment.yaml:180]: These lines are not needed if we apply the change above

```suggestion
```

File: bitnami/minio/templates/gateway/deployment.yaml [hunk 5]
```diff
@@ -167,6 +167,44 @@ spec:
...
+            initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.readinessProbe.successThreshold }}
+            failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
+          {{- else if .Values.customReadinessProbe }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.customReadinessProbe "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.startupProbe.enabled }}
+          startupProbe:
```
Comment (fmulero) [inline bitnami/minio/templates/gateway/deployment.yaml:197]: Same as before:

```suggestion
          startupProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.startupProbe "enabled") "context" $) | nindent 12 }}
```

File: bitnami/minio/templates/gateway/deployment.yaml [hunk 6]
```diff
@@ -167,6 +167,44 @@ spec:
...
+            tcpSocket:
+              port: minio-console
+            initialDelaySeconds: {{ .Values.startupProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.startupProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.startupProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.startupProbe.successThreshold }}
+            failureThreshold: {{ .Values.startupProbe.failureThreshold }}
```
Comment (fmulero) [inline bitnami/minio/templates/gateway/deployment.yaml:204]: These lines can be removed:

```suggestion
```

**Review by fmulero — CHANGES_REQUESTED** [review]
Great!! Thanks a lot @blafry.

I've just request some changes to simplify a little the code. Could you give it a try?

NOTE: Be careful if you accept directly the changes from the github console. All the commits should be signed off

File: bitnami/minio/templates/gateway/deployment.yaml (hunk 1 above)
Comment (blafry) [reply bitnami/minio/templates/gateway/deployment.yaml]: done

File: bitnami/minio/templates/gateway/deployment.yaml (hunk 4 above)
Comment (blafry) [reply bitnami/minio/templates/gateway/deployment.yaml]: done

**Comment by fmulero** [comment]
Sorry @blafry it seems yesterday other changes were merged and our tests failed. Could you resolve the conflicts? I'll merged immediately to avoid this problem again.

**Comment by blafry** [comment]
> Sorry @blafry it seems yesterday other changes were merged and our tests failed. Could you resolve the conflicts? I'll merged immediately to avoid this problem again.

done

**Review by fmulero — APPROVED** [review]
👍

**Pull request merged**